pkg/transport/session/manager.go

```diff
@@ -0,0 +1,106 @@
+// Package session provides a session manager with TTL cleanup.
+package session
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+// Session interface
+type Session interface {
+	ID() string
+	CreatedAt() time.Time
+	UpdatedAt() time.Time
+	Touch()
+}
+
+// Manager holds sessions with TTL cleanup.
+type Manager struct {
+	sessions map[string]Session
+	mu       sync.RWMutex
+	ttl      time.Duration
+	stopCh   chan struct{}
+}
+
+// NewManager creates a session manager with TTL and starts cleanup worker.
+func NewManager(ttl time.Duration) *Manager {
+	m := &Manager{
+		sessions: make(map[string]Session),
+		ttl:      ttl,
+		stopCh:   make(chan struct{}),
+	}
+	go m.cleanupRoutine()
+	return m
+}
+
+func (m *Manager) cleanupRoutine() {
+	ticker := time.NewTicker(m.ttl / 2)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			m.CleanupExpired()
+		case <-m.stopCh:
+			return
+		}
+	}
+}
+
+// AddWithID creates (and adds) a new session with the provided ID.
+// Returns error if ID is empty or already exists.
+func (m *Manager) AddWithID(id string) (Session, error) {
+	if id == "" {
+		return nil, fmt.Errorf("session ID cannot be empty")
+	}
+
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	if _, exists := m.sessions[id]; exists {
+		return nil, fmt.Errorf("session ID %q already exists", id)
+	}
+
+	s := NewProxySession(id)
+	m.sessions[id] = s
+	return s, nil
+}
+
+// Get retrieves a session by ID. Returns (session, true) if found,
+// and also updates its UpdatedAt timestamp.
+func (m *Manager) Get(id string) (Session, bool) {
+	m.mu.RLock()
+	s, ok := m.sessions[id]
+	m.mu.RUnlock()
```

Comment:
I think the touch needs to happen while the lock is being held - otherwise the session could be deleted from the cache before the touch happens.

I also think the map needs to be of type `map[string]*Session` - If the type is a `Session` and not `*Session` then I'm not sure if the value inside the map will be updated with the new timestamp.